Woo: Account details avatar init

lamosty opened the pull request:
In this PR, we add the initial version of the Gravatar and Gravatar Quick Editor integration in the Woo Account details page:

![image](https://github.com/user-attachments/assets/f9ed16bd-f043-4f8f-9bad-c7e1fdeac88b)

It adds your avatar above the Woo account navigation menu and after hovering over the avatar and clicking on it, it opens the quick editor where you can change the avatar.

Props @jcheringer for most of the work/code.

## Testing instructions

Using a local WP site, install Woo. Navigate to `/my-account/edit-account/` and make sure you can see either the default avatar or your based on the user email you have in your local WP setup. If you have some kind of non-existent email or a local one, it will show the default avatar.

lamosty committed:
Woo: Account details avatar init

jcheringer committed:
Woo: Initial QE integration

lamosty committed:
Small refactor

lamosty committed:
Refactor js a bit

lamosty committed:
Follow wp standards

jcheringer committed:
Tweak avatar styles and small clean up

johngodley reviewed (commented):
There are a lot of PHP linting issues. If you haven't set up your editor for the project then it's something you'll need to do.

![image](https://github.com/user-attachments/assets/3e06d5ed-c869-484a-beb3-1b76f62052e7)

johngodley commented on classes/woocommerce/class-account-details-avatar.php:
```diff
@@ -0,0 +1,95 @@
+		add_action( 'woocommerce_after_account_navigation', [ $this, 'end_capture_page' ] );
+	}
+
+	private function get_user_data() {
```
This is a `get` function which doesn't return any data but instead puts the values on member variables, and also performs escaping. I think it might be clearer if the function returned the values, and the escaping can then happen as near to the output as possible.

It may be simpler to just call `wp_get_current_user();` directly when needed as the function doesn't really provide anything extra, with the locale stuff being split into it's own function.

johngodley reviewed (commented):
I'll need to get Woo setup to test this properly, which I'll do later.

Could you also add some details into the readme file?

Question: is it ok this always runs if you have Woo installed - do we need an option or filter to be able to disable it?

johngodley commented:
Ok, everything works as described although the avatar never seems to update on the WP site - it flashes a bit, but stays the same.

One final code comment: `woocommerce_account_details_avatar`. It's quite a long and wordy name, and the others are so short. Could we just use `woocommerce` instead?

lamosty commented:
Thanks for the feedback.

> One final code comment: woocommerce_account_details_avatar. It's quite a long and wordy name, and the others are so short. Could we just use woocommerce instead?

In https://github.com/Automattic/gravatar-enhanced/pull/17/files, I'm doing some refactoring of this function, shortening woocommerce to wc as it's done in Woo. Should I do it here or is it ok if it's done in the follow-up PR? That PR is for a new feature but I thought we should co-locate it inside the `woocommerce` folder.

lamosty committed:
Address feedback

lamosty committed:
Refactor

lamosty committed:
Add readme

lamosty commented:
@johngodley fixed the feedback and also refactored the long name to "wc". Also changed the main name to `my-account` since this works under the whole My Account page, not just the Account Details subpage.

I modified the readme.txt with this new feature, also added it to the bottom changelog. Should I make some screenshot of the feature? How does previewing changes to readme.txt work? Under the Screenshots section, I see only this:
```
== Screenshots ==

1. Gravatar card on profile page
2. Avatar options
3. Gravatar block
```

but no pictures since this is a .txt file. Never developer a plugin before so asking for a friend.

johngodley reviewed (approved):
Thanks, that's all looking great! There's one tab/space issue showing up for me here:

![image](https://github.com/user-attachments/assets/33ee7ac5-09d5-4354-8aee-56c6adb53515)

I've figured out why my avatar wasn't updating - I had the plugin's proxy option enabled. This is a problem that may also affect the other quick editor integration so I've created #18 to track it.

I don't think we necessarily need a screenshot, but if you wanted to do so then it goes in the asset directory.

Other than the formatting issue I think this is good to go.

lamosty commented:
Thanks John!

Ah, I see. I couldn't reproduce the non-updating issue, but didn't try the proxy option. Great that you figured it out.

lamosty committed:
Fix whitespace
